pkg/app/piped/executor/ecs/deploy.go

```diff
@@ -102,7 +102,8 @@ func (e *deployExecutor) ensureSync(ctx context.Context) model.StageStatus {
 		return model.StageStatus_STAGE_FAILURE
 	}
 
-	if !sync(ctx, &e.Input, e.platformProviderName, e.platformProviderCfg, taskDefinition, servicedefinition, primary) {
+	singleton := e.appCfg.QuickSync.Singleton
+	if !sync(ctx, &e.Input, e.platformProviderName, e.platformProviderCfg, singleton, taskDefinition, servicedefinition, primary) {
```

Comment:
Do we need to update for rollback too?

Reply:
I think this feature is only for the sync stage, not progressive, since we only have one task which is available at a time, isn't it?

Reply:
I've forgotten most of the details regarding the code.
At this point, my main concern is to ensure that the rollback process also keeps the singleton property.

Reply:
I got your point 👍 Just plan to update the rollback process logic by another PR, so we can forget about that now.

Reply:
Ok!